.travis.yml

```diff
@@ -1,12 +1,19 @@
 sudo: false
 language: ruby
 rvm:
-  - 2.2
-  - 2.3
-  - 2.4
+- 2.2
+- 2.3
+- 2.4
 before_install:
-  - gem install bundler
+- gem install bundler
 before_script:
-  - psql -c 'create database event_sourcery_test;' -U postgres
+- psql -c 'create database event_sourcery_test;' -U postgres
 addons:
   postgresql: 9.4
+notifications:
+  slack:
+    on_pull_requests: false
+    on_success: change
+    on_failure: always
+    rooms:
+      secure: MNBzPIAQOGxCxM+qV36yYQcrX3RBCKlOAmFP7FbyoavlGA0SHKZc3iuHS+Z23ci/6RD6vCqZNlHNcDosugETUmKRDJBhi3qN/W1HsDtThTT+UaTAYjqZAMYeQVnn/vlH5HGnFfPz75iQe87SkuaP/3y0EKnrJnoHMVR/kItHrv9JcxIsAryf8Lfu3c45TjYmwfRKmiblhoUzTfBpsxSM2s1OWiyQCG4vbjXZj3NIkkhUYC4Sd/Cd0u7BE734z1Uo8UKjuZTBCSyhynqUdXaZCzeDYWpnh27NkyWjl97r4gHNLYYPizpBonJUuX4F1E1Fr+kFGl8qFA1vR9hFLmATWmF3rodagarbAHPQ0/RcpgGW9m7+L6HCTtwLmsNr7ezueTdhYCiLgCO601lRF4fLI2OCGSeSxUYlHfexNZMiqd3/UC6nXGHbWUcTYzdgeTSKggeXiNwnmQYq90pxw7UIZ9r9gNiuBPKHeCOsF2//OynbYABO3bmvbktMsz5sZgl9zgeuXUQqCk6RVANxT+5uZUpPlg8WGsv17fxPmZTxjibk/yEgD6tT4gyIJw2Gn/y5W5WyzCpT8vHU/Dmhj9i/wIHONDLlaDIMO6N/S97UVk8KVhvKkcd//Isbw08HO29Rrv7yq4Hwsb7aIuXyayuEaXhxihz+7kPh1g6imHtXpOg=
```

Comment:
Great question. The Travis docs for notifications are [here](https://docs.travis-ci.com/user/notifications#Configuring-slack-notifications). I used the command:
```
travis encrypt "<account>:<token>#channel" --add notifications.slack.rooms
```
This encrypts the Slack account, token and channel we want the messages sent to.
The encryption uses a public/private key pair. These keys are generated by travis per repo.
The public key is available at this API endpoint: https://api.travis-ci.org/repos/envato/event_sourcery-postgres/key
The private key is kept secret by TravisCI, and is the only party able to decrypt the secret. Hence the answer to your question is TravisCI does the decryption and knows the details to potentially spam us. We're able to disable the integration on Slack's end pretty quickly though.

Note that because a key pair is built per repository, this configuration is not valid in any other project `.travis.yml`. Hence this encrypted test differs from the entry in `event_sourcery`.